driver/controller.go

```diff
@@ -806,11 +807,45 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	return listResp, nil
 }
 
+// ControllerExpandVolume is called from the resizer to increase the volume size.
 func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	d.log.WithField("method", "controller_expand_volume").
-		Info("controller expand volume called")
+	ll := d.log.WithField("method", "controller_expand_volume")
+	ll.Info("controller expand volume called")
+	volID := req.GetVolumeId()
 
-	return nil, status.Error(codes.Unimplemented, "")
+	if len(volID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID missing in request")
+	}
+	volume, _, err := d.storage.GetVolume(ctx, volID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ControllerExpandVolume could not retrieve existing volume: %v", err)
+	}
+	ll.Info("extracting storage from requested capacity range")
+	resizeBytes, err := extractStorage(req.GetCapacityRange())
+	if err != nil {
+		return nil, status.Errorf(codes.OutOfRange, "ControllerExpandVolume invalid capacity range: %v", err)
+	}
+	resizeGigaBytes := resizeBytes / giB
+
+	if resizeGigaBytes <= volume.SizeGigaBytes {
+		// even if the volume is resized independently from the control panel, we still need to resize the node fs when resize is requested
+		// in this case, the claim capacity will be resized to the volume capacity, requested capcity will be ignored to make the PV and PVC capacities consistent 
+		ll.Info("skip volume resizing: current volume size (%v) exceeds requested volume size (%v Gi). Node FS resize will be performed to resize volume claim.", formatBytes(resizeBytes), volume.SizeGigaBytes)
```

Comment:
CI is complaining because `logrus` expects parameters to be attached as key-value results.